pkg/cmd/cli/config/store.go

```diff
@@ -0,0 +1,169 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const (
+	OpenShiftConfigPathEnvVar      = "OPENSHIFTCONFIG"
+	OpenShiftConfigFlagName        = "config"
+	OpenShiftConfigFileName        = ".openshiftconfig"
+	OpenShiftConfigHomeDir         = ".config/openshift"
+	OpenShiftConfigHomeFileName    = ".config"
+	OpenShiftConfigHomeDirFileName = OpenShiftConfigHomeDir + "/" + OpenShiftConfigHomeFileName
+
+	KubeConfigPathEnvVar = clientcmd.RecommendedConfigPathEnvVar
+	KubeConfigFileName   = ".kubeconfig"
+	KubeConfigHomeDir    = ".kube"
+
+	fromFlag     = "flag"
+	fromEnvVar   = "envvar"
+	fromLocalDir = "localdir"
+	fromHomeDir  = "homedir"
+
+	fromKube      = "fromkube"
+	fromOpenShift = "fromopenshift"
+)
+
+type ConfigStore struct {
+	Config           *clientcmdapi.Config
+	Path             string
+	providerEngine   string
+	providerLocation string
+}
+
+func (c *ConfigStore) FromFlag() bool {
+	return c.providerLocation == fromFlag
+}
+
+func (c *ConfigStore) FromEnvVar() bool {
+	return c.providerLocation == fromEnvVar
+}
+
+func (c *ConfigStore) FromLocalDir() bool {
+	return c.providerLocation == fromLocalDir
+}
+
+func (c *ConfigStore) FromHomeDir() bool {
+	return c.providerLocation == fromHomeDir
+}
+
+func (c *ConfigStore) FromOpenShift() bool {
+	return c.providerEngine == fromOpenShift
+}
+
+func (c *ConfigStore) FromKube() bool {
+	return c.providerEngine == fromKube
+}
+
+func GetConfigFromDefaultLocations(clientCfg *client.Config, cmd *cobra.Command) (*ConfigStore, error) {
```

Comment:
Yes, it loads without merging because we want it to actually manage one individual file. Similar to what is done in

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/config/config.go#L68

----- Original Message -----
> > +	return c.providerLocation == fromLocalDir
> > +}
> > +
> > +func (c *ConfigStore) FromHomeDir() bool {
> > +	return c.providerLocation == fromHomeDir
> > +}
> > +
> > +func (c *ConfigStore) FromOpenShift() bool {
> > +	return c.providerEngine == fromOpenShift
> > +}
> > +
> > +func (c *ConfigStore) FromKube() bool {
> > +	return c.providerEngine == fromKube
> > +}
> > +
> > +func GetConfigFromDefaultLocations(clientCfg *client.Config, cmd
> > *cobra.Command) (*ConfigStore, error) {
> 
> godoc.  Is this loading without merging?  I think it is, but I want to be
> sure I'm reading it correctly.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/992/files#r25202539